Headers/Foundation/NSOrderedSet.h

```diff
@@ -0,0 +1,209 @@
+
+/** Interface for NSOrderedSet, NSMutableOrderedSet for GNUStep
+   Copyright (C) 2019 Free Software Foundation, Inc.
+
+   Written by: Gregory John Casamento <greg.casamento@gmail.com>
+   Created: May 17 2019
+
+   This file is part of the GNUstep Base Library.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02111 USA.
+
+*/
+
+#ifndef _NSOrderedSet_h_GNUSTEP_BASE_INCLUDE
+#define _NSOrderedSet_h_GNUSTEP_BASE_INCLUDE
+
+#import <GNUstepBase/GSVersionMacros.h>
+#import <GNUstepBase/GSBlocks.h>
+
+#import <Foundation/NSObject.h>
+#import <Foundation/NSEnumerator.h>
+#import <Foundation/NSIndexSet.h>
+#import <Foundation/NSKeyedArchiver.h>
+
+#if	defined(__cplusplus)
+extern "C" {
+#endif
+  
+@class GS_GENERIC_CLASS(NSArray, ElementT);
+@class GS_GENERIC_CLASS(NSEnumerator, ElementT);
+@class GS_GENERIC_CLASS(NSSet, ElementT);
+@class GS_GENERIC_CLASS(NSDictionary, KeyT:id<NSCopying>, ValT);
+@class NSString;
+@class NSPredicate;
+
+@interface GS_GENERIC_CLASS(NSOrderedSet, __covariant ElementT) : NSObject <NSCoding,
```

Comment:
Should we put this class into something like this?

#if OS_API_VERSION(MAC_OS_X_VERSION_10_7,GS_API_LATEST)
#endif

Other parts of base seem to use this.

Reply:
Added